### PR TITLE
Language lookup helper functions

### PR DESCRIPTION
**Pull request by ivanistheone**
Summary of changes:

  - added `getlang_by_name` for lookup by name, e.g. 'English'
    works for languages with subcodes
     e.g.  "pt-BR":{"name":"Portuguese, Brazil", ..   (comma separated).
     and for multiple lanages semicolon separated
     e.g     "ca":{"name":"Catalan; Valencian", ..
  - Added getlang_by_alpha2 for use in YouTubeSubtitleFile
    using lang lookup strategy inpired by David Hu's TE chef
     https://github.com/fle-internal/te-sushi-chef/blob/master/te_chef.py#L158
  - This PR introduces a dependecy on `pycountry` python package
  - Bumped version number to `0.1.0` in preparation for PyPI release

### CODE REVIEW

**Comment by rtibbles** [comment]
Just to note that English also has subcodes :)

**Comment by ivanistheone** [comment]
`English, Country-where-they-put-vinegar-on-their-fries` ;)

**Comment by rtibbles** [comment]
`English, country where they put gravy on their chips`

File: le_utils/constants/languages.py [hunk 1]
```diff
@@ -52,6 +54,53 @@ def _initialize_language_list():
...
+    Returns None if lookup by interner representation fails.
+    """
     return _LANGLOOKUP.get(code) or None
+    # should this be   return _LANGLOOKUP.get(code) or _LANGLOOKUP[default]  ???
+
+
+_LANGUAGE_NAME_LOOKUP = {l.name: l for l in LANGUAGELIST}
+
+new_items = {}
+for lang_name, lang_obj in _LANGUAGE_NAME_LOOKUP.items():
```
Comment (divad12) [inline le_utils/constants/languages.py]: Nice! Mind adding a small explanatory comment on what this does, e.g. `Add language names that are separated by semicolons, e.g. "Catalan; Valencian"`

File: tests/test_getlangs.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+# TODO: try to match based on language name (stuff before the comma)
+@pytest.mark.skip(reason='De-bracketeded lookup not implemented yet')
+def test_language_names_with_modifier():
+    lang_obj = languages.getlang_by_name('Swahili (macrolanguage)')
+    assert lang_obj is not None, 'Swahili not found'
+    assert lang_obj.code == "sw", 'Wrong internal repr. code'
+    assert lang_obj.name == "Swahili", 'Wrong name'
+    assert lang_obj.native_name == "Kiswahili", 'Wrong native_name'
+
+# TODO: enrich _LANGUAGE_NAME_LOOKUP with individual names from ;-separated lists
```
Comment (divad12) [inline tests/test_getlangs.py]: Seems like this is done now?

File: tests/test_getlangs.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    assert lang_obj is not None, 'Portuguese not found'
+    assert lang_obj.code == "pt", 'Wrong code'
+    assert lang_obj.name == "Portuguese", 'Wrong name'
+    assert lang_obj.native_name == "Português", 'Wrong native_name'
+
+def test_unknown_alpha2_code():
+    lang_obj = languages.getlang_by_alpha2('zz')
+    assert lang_obj is None, 'Uknown code zz returned non-None'
+
+
```
Comment (divad12) [inline tests/test_getlangs.py]: Awesome tests!!

**Comment by ivanistheone** [comment]
@divad12 What use case do you have in mind for `getlang_closest`?

I think something like `getlang_closest_by_name` could be useful (ignore locale info e.g. `zh-Hans`, `Chinese, Mandarin`, `Chinese (mainland)` etc, all match to `zh` = `Chinese`.

Also do you have any ideas how to handle extensible languages? for example a language that we have never seen. I think that one might be out of scope for this PR...

**Comment by divad12** [comment]
@ivanistheone The use case I had in mind was for Ricecooker's new smarter `YouTubeSubtitleFile` that you wrote: https://github.com/learningequality/ricecooker/blob/578a497a2b2cd6f9cb41c37ebfc7753ea148ed8d/ricecooker/classes/files.py#L434

So, e.g. `languages.getlang_closest('en')`, `languages.getlang_closest('zu')`, `languages.getlang_closest('zul')` 

That's OK though, I think what you have is fine.

I'm not sure about a language that we have never seen ... I think you're good to say it's out of scope for now.

**Comment by divad12** [comment]
FYI this is approved. Looks great, merge when ready @ivanistheone

File: le_utils/constants/languages.py [hunk 4]
```diff
@@ -48,16 +59,78 @@ def _initialize_language_list():
...
 
-
 def getlang(code, default=None):
+    """
+    Try to lookup a Language object for `code` in internal representation defined
+    in resources/languagelookup.json.
+    Returns None if lookup by internal representation fails.
+    """
     return _LANGLOOKUP.get(code) or None
+    # should this be   return _LANGLOOKUP.get(code) or _LANGLOOKUP[default]  ???
```
Comment (ivanistheone) [inline le_utils/constants/languages.py]: `default` kwarg doesn't do anything right now. Maybe we could remove it to cleanup the code?

**Pull request merged**